### PR TITLE
Issue #5095 - XmlConfiguration Parser Pool

### LINKED ISSUE #5095: Review XmlConfiguration locking
Spawned by #5083.

**Jetty version**
9.4.x

**Description**
`XmlConfiguration` has a `static` instance of `XmlParser` and in the constructor locks on it and performs the parsing.

This is fine from the locking point of view, but it grabs a JVM-wide lock so that `XmlConfiguration`s can only execute serially, which may not be desirable and may slow down startup.

The JVM-wide `XmlParser` instance was created once for performance reasons due to costly initialization?
Can we do better?

### PR DESCRIPTION
**Pull request by gregw**
For #5095

### CODE REVIEW

File: jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java [hunk 1]
```diff
@@ -99,7 +100,7 @@
             ArrayList.class, HashSet.class, Queue.class, List.class, Set.class, Collection.class
         };
     private static final Iterable<ConfigurationProcessorFactory> __factoryLoader = ServiceLoader.load(ConfigurationProcessorFactory.class);
-    private static final XmlParser __parser = initParser();
+    private static final Pool<ConfigurationParser> __parsers = new Pool<>(Runtime.getRuntime().availableProcessors(),1);
```
Comment (lorban) [inline jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java]: I would cap that `Runtime.getRuntime().availableProcessors()` up to a reasonable number (8? 16? 32?) as I don't think we could use 128+ parsers (which is becoming a more and more common HW thread count) in parallel in practice. Or could we?

**Review by lorban — APPROVED** [review]
LGTM besides that open question of mine.

File: jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java (hunk 1 above)
Comment (gregw) [reply jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java]: Good idea.  Considering we were previously contending on 1 instance, I think a limit of 8 would be fine.

**Comment by gregw** [comment]
@sbordet bump.  I need a green tick from you (until such time as @lorban is a committer)

**Comment by gregw** [comment]
@sbordet @janbartel somebody? anybody????   NUDGE!!!!!

**Review by joakime — APPROVED** [review]
LGTM

**Pull request merged**